crates/win-api-wrappers/src/service.rs

```diff
@@ -0,0 +1,150 @@
+use std::alloc::Layout;
+
+use windows::core::Owned;
+use windows::Win32::Foundation::{ERROR_INSUFFICIENT_BUFFER, GENERIC_READ};
+use windows::Win32::System::Services::{
+    OpenSCManagerW, OpenServiceW, QueryServiceConfigW, QueryServiceStatus, StartServiceW, QUERY_SERVICE_CONFIGW,
+    SC_HANDLE, SC_MANAGER_ALL_ACCESS, SERVICE_ALL_ACCESS, SERVICE_AUTO_START, SERVICE_BOOT_START, SERVICE_DEMAND_START,
+    SERVICE_DISABLED, SERVICE_QUERY_CONFIG, SERVICE_QUERY_STATUS, SERVICE_RUNNING, SERVICE_STATUS,
+    SERVICE_SYSTEM_START,
+};
+
+use crate::raw_buffer::RawBuffer;
+use crate::utils::WideString;
+
+pub struct ServiceManager {
+    handle: Owned<SC_HANDLE>,
+}
+
+impl ServiceManager {
+    pub fn open_read() -> anyhow::Result<Self> {
+        Self::open_with_access(GENERIC_READ.0)
+    }
+
+    pub fn open_all_access() -> anyhow::Result<Self> {
+        Self::open_with_access(SC_MANAGER_ALL_ACCESS)
+    }
+
+    fn open_with_access(access: u32) -> anyhow::Result<Self> {
+        // SAFETY: FFI call with no outstanding preconditions.
+        let raw_sc_handle = unsafe { OpenSCManagerW(None, None, access)? };
+
+        // SAFETY: Handle returned by `OpenSCManagerW` is valid and needs to be closed after use,
+        // thus it is safe to take ownership of it via `Owned`.
+        let handle = unsafe { Owned::new(raw_sc_handle) };
```

Comment:
nitpick: "… thus it is safe to do …" is generally not useful and typically could be removed. A SAFETY: comment, by definition, is here "**to explain why the unsafe block is safe**, and which invariants are used and must be respected". It’s important to elaborate enough, but let’s try to keep the wording minimal.

suggestion:
```suggestion
        // SAFETY: On success, the handle returned by `OpenSCManagerW` is valid and owned by the caller.
        let handle = unsafe { Owned::new(raw_sc_handle) };
```